shared/src/library/kotlin/org/jetbrains/kotlin/konan/library/SearchPathResolver.kt

```diff
@@ -93,7 +96,9 @@ internal class KonanLibraryProperResolver(
         val resolverTarget = this.target
         val candidatePath = candidate.libraryFile.absolutePath
 
-        if (!candidate.targetList.contains(resolverTarget.visibleName)) {
+        if (!candidate.targetList.contains(resolverTarget.visibleName) &&
+                (candidate.targetList.isNotEmpty() || !allowEmptyTargets)
```

Comment:
This change has been superseded by changes made by @alexander-gorshenev, also introducing support for target-less klibs.
As to the suggestion, I don't know if that would be a better solution and could use some advice. To me, it seems that the fact that a library only contains the frontend metadata shows that it shouldn't have any targets at all, but I might not have the right understanding.

Reply:
Introduction of the new special target involves new special entries for `Family` and `Architecture` enums as well, which will definitely pollute target-specific code (e.g. `when` over these entities).

`DESCRIPTORS_ONLY_LIBRARY` also looks like a hack, but I think it is a lesser evil.

It may be better to turn `CompilerOutputKind` into a sealed class and make `LIBRARY` a class with a property that describes this library, e.g. that it has a `KonanTarget` or it only contains the metadata, although it require additional refactoring.

Reply:
We've discussed it all a little within the native team. Let's have the same "-produce library" for these artifacts. But let's have a -X... flag asking to  produce metadata only klibs.

Reply:
Also no need to mess with any new targets in this case. Let's leave to until we have a common backend.

Reply:
Ok, I'll rework this. @alexander-gorshenev Do you think `CommonCompilerArguments.klibBasedMpp` is suitable for such a flag, or should there be another one that is more specific to frontend-only libraries?

Reply:
I'd say let's have separately "-Xklib-mpp" and "-Xmetadata-klib". 
But they both can turn on the same AnalysisFlags.klibBasedMpp which should be renamed to something like AnalysisFlags.noSingleModuleChecks for now.

Reply:
I've force-pushed the branch, it now has the `-Xmetadata-klib` in `K2NativeCompilerArguments`. I'm not setting the analysis flag that disables exect-actual checks based on it, since it seems to me that it's OK to set it only from `-Xklib-mpp` in the shared code and pass both arguments when compiling shared-Native source sets.